core/resource.go

```diff
@@ -18,22 +23,43 @@ import (
 	reflectpb "google.golang.org/grpc/reflection/grpc_reflection_v1alpha"
 )
 
+// BasePath define path where proto file will persisted
+const BasePath = "/tmp/grpcox/"
+
 // Resource - hold 3 main function (List, Describe, and Invoke)
 type Resource struct {
 	clientConn *grpc.ClientConn
 	descSource grpcurl.DescriptorSource
 	refClient  *grpcreflect.Client
+	protos     []Proto
 
 	headers []string
 	md      metadata.MD
 }
 
 //openDescriptor - use it to reflect server descriptor
-func (r *Resource) openDescriptor() {
+func (r *Resource) openDescriptor() error {
 	ctx := context.Background()
 	refCtx := metadata.NewOutgoingContext(ctx, r.md)
 	r.refClient = grpcreflect.NewClient(refCtx, reflectpb.NewServerReflectionClient(r.clientConn))
-	r.descSource = grpcurl.DescriptorSourceFromServer(ctx, r.refClient)
+
+	// if no protos available use server reflection
+	if r.protos == nil {
+		r.descSource = grpcurl.DescriptorSourceFromServer(ctx, r.refClient)
+		return nil
+	}
```

Comment:
grpcox is expected to use by multiple services at the same time, with this script served on single server... so with your current code, if there's one service request using local proto, then another service will also use that proto request instead of reflection to server... 

and you also need to think how if multiple service request with local protos, you need to mapping proto files based on host..

 my suggestion, instead of you modify `getLists` handler with `getListsWithProto` ... better to make new endpoint called register proto file, that endpoint will get paramter host, and protofile, then will store proto file locally mapped to host ... after that the rest functionality with remain same... then in this function you need to check whether requested host has proto file mapped? if yes then use `DescriptorSourceFromProtoFiles`.

wdyt?

Reply:
use the same way as I mapped connection object based on host

Reply:
oh I just read that you already mapped protos based on host target. ok then the problem is only functionality to handle request with proto and server reflection at the same time.

Reply:
I see, so we should have to use the local proto only for the request that explicitly ask for it? Where do you think we should implement that?

Reply:
I think you can mapping proto file based on host, so if there's no proto file for that host, go to server reflection.. wdyt?

Reply:
I see, I'm gonna work on this then

Reply:
oh sorry my bad, long time not reading my code haha... oke approved, lgtm

Reply:
I see. Thanks, for taking time reviewing my PR. Really appreciate it :)